lib/KettleMiddleware-Multer.js

```diff
@@ -0,0 +1,229 @@
+
+    return multerMethod === "array" ?  multer[multerMethod](multerFields, formFieldOptions.maxCount) : multer[multerMethod](multerFields);
+};
+
+fluid.defaults("kettle.middleware.multer.filter", {
+    gradeNames: ["fluid.component"],
+    // This must be a filter function in Multer's
+    // expected style - see https://github.com/expressjs/multer#filefilter
+    //
+    // See the mimeType filter for an example of using argument-shifting
+    // to supply configuration material while maintaining Multer's expected
+    // signature
+    invokers: {
+        multerFileFilter: {
+            funcName: "fluid.notImplemented"
+        }
+    }
+});
+
+fluid.defaults("kettle.middleware.multer.filter.allowAll", {
+    gradeNames: ["fluid.component"],
+    invokers: {
+        multerFileFilter: {
+            funcName: "kettle.middleware.multer.filter.allowAll.filterFunc"
+        }
+    }
+});
+
+// Duplicate of Multer's internal allowAll function;
+// Multer doesn't export, making it inaccessible
+// to the kettle.npm.multer namespace
+
+kettle.middleware.multer.filter.allowAll.filterFunc = function (req, file, cb) {
+    cb(null, true);
+};
+
+fluid.defaults("kettle.middleware.multer.filter.mimeType", {
+    gradeNames: ["fluid.component"],
+    // an array of mimetypes to accept, FE:
+    // acceptedMimeTypes: ["image/gif", "image/jpg", "image/gif"],
+    acceptedMimeTypes: [],
+    invokers: {
+        multerFileFilter: {
+            funcName: "kettle.middleware.multer.filter.mimeType.filterFunc",
+            args: ["{arguments}.0", "{arguments}.1", "{that}.options.acceptedMimeTypes", "{arguments}.2"]
+        }
+    }
+});
+
+// Filters by mimeType based on a list of accepted mime types
+kettle.middleware.multer.filter.mimeType.filterFunc = function (req, file, acceptedMimeTypes, cb) {
+    var isAcceptableMimeType = fluid.contains(acceptedMimeTypes, file.mimetype);
+    cb(null, isAcceptableMimeType);
+};
+
+// Abstract grade for storage
+fluid.defaults("kettle.middleware.multer.storage", {
+    gradeNames: ["fluid.component"],
+    // This must be one of the supported Multer
+    // storage functions (memoryStorage or diskStorage,
+    // unless using additional plug-ins for Multer);
+    // see the concrete grades below
+    invokers: {
+        multerStorage: {
+            funcName: "fluid.notImplemented"
+        }
+    }
+});
+
+// Storage grade using Multer's memoryStorage
+// storage function - uploaded files are only
+// stored in memory for the duration of the
+// request, although a handler or other
+// middleware can process them from
+// request.req.file or request.req.files
+fluid.defaults("kettle.middleware.multer.storage.memory", {
+    gradeNames: ["kettle.middleware.multer.storage"],
+    invokers: {
+        multerStorage: {
+            funcName: "kettle.npm.multer.memoryStorage",
+            args: []
+        }
+    }
+});
+
+// Storage grade using Multer's diskStorage
```

Comment:
These comments have received painfully short line lengths - I think the community considered that 110 or so was a reasonable line length

Reply:
I normally break at 80 characters. Not sure what was going on to make me decide that 40-50 was more reasonable here. Fixed.